### PR TITLE
Adapt Flyway as Registry DB migration tool

### PR DESCRIPTION
**Pull request by yhfreeman**
- [x] add migration SQL files following flyway naming conventions based on our existing schema / reference data build-out

- [x] make flyway work locally

- [x] make flyway work on openshift

- [x] adapt the existing GH actions - oc patterns and update / build flyway image upon changes on migration SQL files

- [x] [documenting a plan to integrate this tool with existing prod db with zero downtime](https://github.com/bcgov/platform-services-registry/issues/199#issuecomment-750601880)

- [x] minor secret name fix on file openshift/backup/deploy-values.yaml

Due to lengthy sql file transfers:
/bot-ignore-length

### CODE REVIEW

**Comment by yhfreeman** [comment]
converted back to a draft in order to add the following:

- [ ] remove the dev build process of github action (as we are not sure if we should service account the privilege of building from an oc template)

- [ ] add a comment so we know the out-of-box docker image being used on oc is under root user (something we can fix later)

- [ ] encapsulate app label name in oc templates as groundwork (to label temporary artifacts)

- [ ] update the implementation scripts for migrating this in prod in temp_flyway_migration folder

- [ ] update repo README for the new db, api deployment, nsp to reflect this PR

File: openshift/templates/nats.yaml [hunk 1]
```diff
@@ -100,4 +100,11 @@ parameters:
...
-    value: registry-nats
\ No newline at end of file
+    value: registry-nats
+  - displayName: App label name
+    description: |
+      The App label name used for all components of the application. This is useful
+      in dev and test when you want to clear all and have a clean working namespace.
+    name: APP_LABEL_NAME
+    required: true
+    value: platsrv-registry
```
Comment (sbathgate) [inline openshift/templates/nats.yaml:110]:
```suggestion
    value: platsrv-registry
    
```

File: openshift/templates/nsp.yaml [hunk 2]
```diff
@@ -140,4 +155,15 @@ parameters:
...
+    description: |
+      The Aporeto identifier used by Aporeto to identify flyway pod.
+    required: true
+  - displayName: App label name
+    description: |
+      The App label name used for all components of the application. This is useful
+      in dev and test when you want to clear all and have a clean working namespace.
+    name: APP_LABEL_NAME
+    required: true
+    value: platsrv-registry
```
Comment (sbathgate) [inline openshift/templates/nsp.yaml:169]:
```suggestion
    value: platsrv-registry
    
```

File: temp_flyway_migration/db-deploy.yaml [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+    value: netapp-file-standard
+  - name: IMAGE_REGISTRY
+    value: image-registry.openshift-image-registry.svc:5000
+  - displayName: App label name
+    description: |
+      The App label name used for all components of the application. This is useful
+      in dev and test when you want to clear all and have a clean working namespace.
+    name: APP_LABEL_NAME
+    required: true
+    value: platsrv-registry-flyway
```
Comment (sbathgate) [inline temp_flyway_migration/db-deploy.yaml:343]:
```suggestion
    value: platsrv-registry-flyway
    
```

File: temp_flyway_migration/nsp.yaml [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    description: |
+      The Aporeto identifier used by Aporeto to identify flyway pod.
+    required: true
+  - displayName: App label name
+    description: |
+      The App label name used for all components of the application. This is useful
+      in dev and test when you want to clear all and have a clean working namespace.
+    name: APP_LABEL_NAME
+    required: true
+    value: platsrv-registry-flyway
```
Comment (sbathgate) [inline temp_flyway_migration/nsp.yaml:92]:
```suggestion
    value: platsrv-registry-flyway
    
```

**Review by sbathgate — APPROVED** [review]
Mostly just nitpicks. As discussed in RC, a minor addendum to the README to include something along the lines of:
> FLYWAY_PASSWORD == APP_DB_PASSWORD
> FLYWAY_USER == APP_DB_USER
> FLYWAY_PLACEHOLDERS_USERNAME == APP_DB_USER

File: db/openshift/flyway.build.yaml [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+      type: Git
+    strategy:
+      dockerStrategy:
+        dockerfilePath: flyway.Dockerfile
+parameters:
+- description: |
+    The name assigned to all of the objects defined in this template.
+    You should keep this as default unless your know what your doing.
+  displayName: Name
+  name: NAME
```
[the remark this answers is not in the document]
Comment (yhfreeman) [reply db/openshift/flyway.build.yaml:51]: This bc template will be used in tools namespace, so slightly different from all the other templates that we modified with the APP_LABEL_NAME. But you are right, we could choose to update all the manifests used in tools namespace with some label name.

**Comment by yhfreeman** [comment]
> Mostly just nitpicks. As discussed in RC, a minor addendum to the README to include something along the lines of:
> 
> > FLYWAY_PASSWORD == APP_DB_PASSWORD
> > FLYWAY_USER == APP_DB_USER
> > FLYWAY_PLACEHOLDERS_USERNAME == APP_DB_USER

Didn't see anything in our README about local dev setup that refers to these existing vars (APP_DB_PASSWORD, APP_DB_USER). Perhaps we could make sure to add those lines when we further develop our existing documentation. Thanks.

**Pull request merged**